examples/add_particles_numpy/add_particles_numpy.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+
+import espressopp
+from espressopp import Real3D, Int3D
+from espressopp.tools import decomp, lattice, velocities
+
+from mpi4py import MPI
+import time
+import numpy as np
+
+def generate_particles(particles_per_direction):
+    num_particles = particles_per_direction**3
+    x, y, z, Lx, Ly, Lz = lattice.createCubic(num_particles, rho=0.8442, perfect=False)
+    vx, vy, vz = velocities.gaussian(T=0.6, N=num_particles, zero_momentum=True)
+    return x, y, z, Lx, Ly, Lz, vx, vy, vz
+
+def generate_system(add_particles_array):
+    rc = 2.5
+    skin = 0.3
+    timestep = 0.005
+    temperature = 1.0
+    comm = MPI.COMM_WORLD
+
+    particles_per_direction = 64
+    x, y, z, Lx, Ly, Lz, vx, vy, vz = generate_particles(particles_per_direction)
+
+    num_particles = len(x)
+    density = num_particles / (Lx * Ly * Lz)
+    size = (Lx, Ly, Lz)
+
+    system = espressopp.System()
+    system.rng = espressopp.esutil.RNG()
+    system.bc = espressopp.bc.OrthorhombicBC(system.rng, size)
+    system.skin = skin
+    nodeGrid = decomp.nodeGrid(comm.size, size, rc, skin)
+    cellGrid = decomp.cellGrid(size, nodeGrid, rc, skin)
+    system.storage = espressopp.storage.DomainDecomposition(system, nodeGrid, cellGrid)
+
+    if add_particles_array:
+
+        assert(system.storage.hasAddParticlesArray())
+
+        tstart = time.time()
+        props = ['id', 'type', 'mass', 'posx', 'posy', 'posz', 'vx', 'vy', 'vz']
+        ids      = np.arange(1,num_particles+1)
+        types    = np.zeros(num_particles)
+        mass     = np.ones(num_particles)
+        x        = np.array(x)
+        y        = np.array(y)
+        z        = np.array(z)
+        vx       = np.array(vx)
+        vy       = np.array(vy)
+        vz       = np.array(vz)
```

Comment:
We could use two-dimensional arrays to save some parameters. At the same time it ensures that all three components are present.

Reply:
Problem is not all properties are 3d so we can't build a single numpy array from that. Also the x,y,z arrays are already read/generated directly from `lammps.read(...)` and `tools.readxyz(...)` as separate arrays.

BTW, I've removed these lines from the example since np.stack already treats them as arrays.